### DISCUSSION TITLE
Create access token for organization

### DISCUSSION THREAD

**Discussion by eamonburns** [community/community#74701] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

How do I create a personal access token for my organization, so that I can read/clone a private repository?
I can see that I can create a PAT policy in "Organization > Settings > Personal Access Tokens" but I don't see where I can make (or request) a new one

**Comment by gitstua** [comment, marked as answer]
Hi @Agent-E11,

Personal Access Tokens are created for the user and can be assigned permissions to your repos and organizations.

The to generate a PAT, follow these steps:

1. Navigate to your personal settings, by clicking on your profile photo in the upper right corner and selecting Settings.
2. In the left sidebar, click Personal access tokens.
3. Click **Generate** new token.
4. In the "Note" field enter a descriptive name for your token.
5. To set an expiration for your token, select Expiration, then choose a default option or click Custom to enter a date.
6. Select the scopes you'd like to grant this token. Keep in mind, to use your token to access repositories from the command line, you should select repo.
7. Click **Generate token**.
8. Optionally, click the clipboard icon to copy the new token 

Hope this helps!

**Reply by eamonburns** [reply]
One other thing is that you need to the select the organization as the "Resource owner" in the "Generate new token" page.

**Reply by jtosey** [reply]
I did step (2) but this is a BETA drop-down menu for Settings, Active Tokens and Pending requests. I don't see an option for Generate a new token.

**Reply by suchismita1980** [reply]
About step 2: I am not able to see "Personal access tokens" tab in left side bar.
<img width="133" alt="image" src="https://github.com/community/community/assets/48068818/da312346-207b-4a15-aa83-ce98181291fd">

**Reply by eamonburns** [reply]
@suchismita1980 For me, I can click "Developer Settings" at the bottom of the left side-panel
<img width="242" alt="image" src="https://github.com/community/community/assets/116777388/694bda16-db96-4311-8fad-8c2d0966f8c8">

Then click "Personal access tokens > Fine-grained tokens"
<img width="233" alt="image" src="https://github.com/community/community/assets/116777388/5620b996-81c1-46c2-b2e5-a7d0b9fd53a7">

Then you can click "Generate new token" in the top right

**Reply by trusktr** [reply]
But how do we make the token apply only to a repo in an org? When I try to select a repo, I do not see the org repo in the list!

EDIT: Doh! That's what the "Resource owner" option is for!

**Reply by pmverma** [reply]
There is no "Resource owner" option anymore!!

**Reply by battlesnake** [reply]
Github UX makes me miss AWS Console

**Reply by alelom** [reply]
> There is no "Resource owner" option anymore!!

Yes there is (leaving this comment for future me, mostly):

![image](https://github.com/user-attachments/assets/34fefc0e-c537-4c37-9598-abf34d204161)

**Reply by Malix-Labs** [reply]
Not for classic tokens.

**Comment by eamonburns** [comment]
It did help, going through the steps (and knowing that it was possible) made me realize I forgot to select the _organization_ instead of _my username_ under "Resource owner".
Thank you

**Reply by cbm755** [reply]
probably 1.5 hours trying again and again, never even noticed the "resource owner", rather embarrassed ;-)

**Reply by shawnwall** [reply]
for me for some reason, only my name is under resource owner, and none of the organizations i belong to or admin

**Reply by shawnwall** [reply]
nevermind, had to go and set up access for the organization to allow them.

**Reply by Aetherinox** [reply]
Honestly this shouldn't be this difficult. Under organization settings where you enable access tokens for an organization, they should have just added the ability to create a PAT from that section, and then list it in the organization. Not having to run back to your personal settings and then tweak the owner.

**Comment by mr-kushan** [comment]
Use Developer settings.

**Comment by drcongo** [comment]
I realise this is an old and "answered" thread, but the option to select my organisation as the "Resource owner" has disappeared from my personal access token (classic) create screen. How do I make it come back?

This is what I see in the org settings at `/organizations/my-org/settings/personal-access-tokens?tab=classic`...
![Screen Shot 2024-12-09 08 54 38 PM](https://github.com/user-attachments/assets/d5056116-40ea-4f7a-8850-18f2e5b898ae)

And this is what I see on the generate token screen...
![Screen Shot 2024-12-09 08 56 35 PM](https://github.com/user-attachments/assets/fc47b580-0544-4ebf-a966-63f571e3c94b)

**Reply by pmverma** [reply]
I am looking for the same!

**Reply by KonnorRogers** [reply]
Same. Hitting this problem today.

**Reply by drcongo** [reply]
@gitstua Is it possible to get the `Answered` tag removed from this? Seems something has changed on GitHub.

**Reply by IAFahim** [reply]
Might be here https://github.com/settings/tokens
It might be moved to personal access token. I am testing it right now
Yeapp https://github.com/settings/personal-access-tokens/new you can select organization from there

**Reply by drcongo** [reply]
@IAFahim That's only for the fine grained access tokens as far as I can see, I need it for the classic tokens.

**Reply by IAFahim** [reply]
Yes Sir @drcongo, first Link:
https://github.com/settings/tokens
![image](https://github.com/user-attachments/assets/dcc6f29d-5f7b-460b-a739-fc39b7e26f20)

**Reply by drcongo** [reply]
@IAFahim And then you click on `Generate new token`, and choose `Generate new token (classic)` from the dropdown, and then you end up on a screen where it's impossible to choose an organisation. Which is exactly the problem we're asking about.

![Screen Shot 2025-01-13 04 24 22 PM](https://github.com/user-attachments/assets/1a3906ee-e2a0-4984-947e-31151c1a36f0)

**Reply by IAFahim** [reply]
For classic I didn't check.
But for fine grain you can
![Screenshot_2025-01-14-03-00-20-42_3aea4af51f236e4932235fdada7d1643](https://github.com/user-attachments/assets/26fadc01-9856-48d2-be0a-d76995786482)

**Reply by drcongo** [reply]
We know that, we're asking where it's gone for classic.

**Reply by jflamy** [reply]
I have the exact same problem.

**Reply by EspressoKris** [reply]
I came across this problem and realised that my organisation wasn't set up.

Access your **organisation**, then go to **settings**, scroll down to the **third-party Access** tab and click on **Personal-access tokens**. Complete the configuration and **enrol your organisation** and you will now find it under Resource owner in the fine-grained tokens page.

**Reply by drcongo** [reply]
It's not the fine-grained tokens we want / need, it's the classic tokens. 

Every time I see there's been activity in this thread I get excited only to find out that it's people suggesting the fine-grained tokens again. 😭

**Reply by excalq** [reply]
We're stuck with classic tokens until FG-PATs support GH (NPM) package registry.

**Comment by davidt-gh** [comment]
So there is no option to create a global token? what if I want to create a github token to make calls between GitHub actions in organization but the user leave the organization after one year? what then?

**Reply by karfau** [reply]
One way to workaround this is to create (and pay for) a company owned github user account with the credentials available to the admins/people responsible for managing such secrets.

**Comment by tahoward** [comment]
Just ran into this as well, I just want classic tokens for my organization. 😩

**Comment by ozkary** [comment]
I was able to create an organization token by using the fine-grained token options. Under the owner, select the organization not your name. We also need to set the permissions.

It is getting harder with all these platform's outdated documentation. Things are done one way, then moved around, leaving us with some older documentation.

**Reply by drcongo** [reply]
Everybody is able to create an org-scoped _fine-grained_ token. This thread is specifically about it being missing from the _classic_ tokens. Some of us _need_ classic tokens still.

**Comment by Tectract** [comment]
How do I create a personal access token for my organization, so that I can read/clone a private repository?

Can anyone give me a straight answer here?

what in the literal FUCK. HOW DO I CLONE A REPO??? 

do not give me some bullshit about an APP. Why the F am I supposed to use an app to CLONE MY REPO??? THIS IS SUPPOSED TO BE A COMMAND LINE COMMAND YOU IDIOTS.

what is the URL to set the token for my org???

What are the command-line commands to clone a repo, for my org???

**Reply by Aetherinox** [reply]
For **Fine-grained Tokens**, you have to go to your personal account access tokens at: https://github.com/settings/personal-access-tokens. 

To get there manually, click your avatar in the top-right of Github. Select **Settings**

![image](https://github.com/user-attachments/assets/cb783688-9e24-4319-9166-68f4f8293747)

Then on the left, go down to **Developer Settings**

![image](https://github.com/user-attachments/assets/1be3871d-4285-4a26-b19a-bd17572fcb63)

Then on the left-side menu, click **Fine-Grained Token**.

![image](https://github.com/user-attachments/assets/f5bb50ab-9733-44a9-87e2-2f47f5e0aeb0)

Click the "Create New Token" button at the top right.

On the page where you provide the token details, scroll down to **Resource Owner** and click the dropdown. Select you org

![3mEKFuPGcQ](https://github.com/user-attachments/assets/e2adab8f-dcb6-4fc8-8cee-762f3065543e)

Select your Org, in my example, it's **Test Org**.

The page will refresh, and then you can input the token info.

Github also has **Classic Tokens**, available at the URL https://github.com/settings/tokens

You cannot create an organization classic token from that page. The tokens are for your personal account. You want the one I explained above, **Fine-grained Token**

**Reply by Tectract** [reply]
this is the stupidest bs ever. github is becoming impossible to use now that MS took over

**Reply by Abhishek-More** [reply]
valid crashout

**Reply by levelupsec** [reply]
@drcongo 

It looks like classic tokens will Just Work(TM) on organizations your account has access to. `git clone https://<classic token>@github.com/<org>/<repo>.git`. You don't actually _need_ to set the org for a classic. So long as your account has access to the org, the classic token will work on it

**Reply by KonnorRogers** [reply]
> @drcongo
> 
> It looks like classic tokens will Just Work(TM) on organizations your account has access to. `git clone https://<classic token>@github.com/<org>/<repo>.git`. You don't actually _need_ to set the org for a classic. So long as your account has access to the org, the classic token will work on it

But doesn't work when you try to use the token in GH actions since its not associated to a user.

**Reply by cognivore** [reply]
> valid crashout

I'm in the middle of one right now and I'm happy that I don't have to type it all out.

Thank you @Tectract!

**Reply by cognivore** [reply]
Namely — putting them under "Developer settings"... 

...as opposed to what other settings? "Project manager settings"? Developers are like 99.99% of users of this website. 🤡

**Comment by STS-SW** [comment]
Hi everyone, 
sorry for bringing this topic back up. 
I'm trying to upload a NuGet package to my **organization**'s NuGet repository. 
What kind of token should I use, and with which permissions?
To provide more context:

- I'm not using GitHub Actions — I'm trying to publish the package from the powershell command line
- The user generating the token is an administrator of the organization
- The repositories in the organization are private

Thanks in advance. 
Cheers, Stefano

**Comment by reggie319** [comment]
I need to create a PAT (Person Access Token -Fine grained) to upload NuGet packages to my Organization.
I created the PAT but I don't see/know which permission to use that allows Admin access to the NuGet packages.

Thanks

**Reply by STS-SW** [reply]
A -> Choose the resource owner, here you can find your company
B -> Choose the token expiration
C -> Choose the repositories
D -> Choose the permissions

<img width="1175" height="1458" alt="image" src="https://github.com/user-attachments/assets/c60c4b38-8682-445b-bfe7-ef7f560c708e" />

**Reply by reggie319** [reply]
But this token doesn't have the ability to manage NuGet packages (that I can see).
I need NuGet packages to be part of my Organization (Fine-Grain), not at the User level like PAT (classis) does.

**Comment by Tectract** [comment]
It's still impossible to even FIND the "fine-grained access tokens" part of this site. Every time I have to deal with it, I spit curses out of my mouth for github making us use this idiotic thing. All my keys are named "F_GITHUB" lol.
